lib/hooks/LocalStorage/index.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+
+/**
+Hook to store a persistent state in local storage.
+
+The state is shared among any component using it
+from any tab of the same browser
+accessing any application hosted in the same domain.
+
+A null value means the value is unset.
+Setting the state to null unsets it.
+Any other value is converted to string including undefined.
+The initial value is set only if the value is unset
+on component initialization.
+Initial value of undefined is ignored.
+*/
+
+export default function useLocalStorage(key, initialValue) {
+  function decode(json) {
+    const value = json === null ? null : JSON.parse(json)
+    return value
+  }
+
+  function encodeValue(value) {
+    const json = value === null ? null : JSON.stringify(value)
+    return json
+  }
+
+  function setStorage(value) {
+    if (value === null) localStorage.removeItem(key)
+    else localStorage.setItem(key, encodeValue(value))
+  }
+
+  const [state, setState] = React.useState(() => {
+    const storedValue = localStorage.getItem(key)
+    if (storedValue === null) {
+      if (initialValue !== undefined) {
+        setStorage(initialValue)
+      }
+    }
+    return decode(localStorage.getItem(key))
+  })
+  // Update state when changed from other tab
+  React.useEffect(() => {
+    const listener = (e) => {
+      if (e.key === key) {
+        setState(decode(e.newValue))
+      }
+    }
+    window.addEventListener('storage', listener)
+    return () => {
+      window.removeEventListener('storage', listener)
+    }
+  }, [])
+  // Offer a setter that updates the storage
+  // The state is set to the actual storage value (str conversion or null)
+  const persistentSetState = React.useCallback((value) => {
+    setStorage(value)
+    setState(value)
+  })
+  return [state, persistentSetState]
```

Comment:
Do you think memoize the function with key as a dependency could be worth?
Example:
```javascript
  const persistentSetState = React.useCallback(
    (value) => {
      setStorage(value)
      setState(value)
    },
    [key] // Memoize the function with the key as a dependency
  )
```

Reply:
I think that the key is not needed as dependency as each call to the hook will hold its own useCallback hook. Why do you think such a think is necessary?

Reply:
Why use useCallback without dependencies? If you don't have it depending on anything it will be re-created every time, as if you haven't put it.

Reply:
Just the first time, i just need it to be the same for every call to avoid redraws when passing it as attribute.